internal/controller/dnsrecord_controller.go

```diff
@@ -220,25 +221,34 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	if err != nil {
 		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse,
 			"DNSProviderError", fmt.Sprintf("The dns provider could not be loaded: %v", err))
-		return r.updateStatus(ctx, previous, dnsRecord, false, []string{}, err)
+		return r.updateStatus(ctx, previous, dnsRecord, nil, false, []string{}, err)
 	}
 
+	probes := &v1alpha1.DNSHealthCheckProbeList{}
 	if probesEnabled {
 		if err = r.ReconcileHealthChecks(ctx, dnsRecord, allowInsecureCert); err != nil {
 			return ctrl.Result{}, err
 		}
+		// get all probes owned by this record
+		if err := r.List(ctx, probes, &client.ListOptions{
```

Comment:
would there ever be more than one for a given record

Reply:
Yes, a probe per "leaf". If you have two addresses on your gateway you will end up with an A record with two IPs and you will have a probe per IP